jetty-websocket/websocket-core-server/src/main/java/org/eclipse/jetty/websocket/core/server/internal/UpgradeHttpServletRequest.java

```diff
@@ -400,13 +406,17 @@ public String getRemoteHost()
     @Override
     public void setAttribute(String name, Object value)
     {
-        throw new UnsupportedOperationException(UNSUPPORTED_WITH_WEBSOCKET_UPGRADE);
+        if (request == null)
+            throw new UnsupportedOperationException(UNSUPPORTED_AFTER_WEBSOCKET_UPGRADE);
```

Comment:
This is not what the OP requested.  If request is null, then the attribute needs to be removed from the copy of the attributes.